include/modem/location.h

```diff
@@ -0,0 +1,269 @@
+		/** Configuration for LOC_METHOD_GNSS. */
+		struct loc_gnss_config gnss;
+		/** Configuration for LOC_METHOD_WIFI. */
+		struct loc_wlan_config wlan;
+	};
+};
+
+/** Location request configuration. */
+struct loc_config {
+	/** Number of location methods in 'methods'. */
+	uint8_t methods_count;
+	/**
+	 * @brief Selected positioning methods and associated configurations in priority order.
+	 *
+	 * @details Index 0 has the highest priority. Number of methods is indicated in
+	 * 'methods_count'.
+	 */
+	struct loc_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
+	 */
+	uint16_t interval;
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct loc_event_data *event_data);
+
+/** @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ *          configuration, a single position or periodic position updates are given. The results are
+ *          delivered to the event handler function.
+ *
+ *          Periodic position updates can be stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request(const struct loc_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets default values to given configuration.
+ *
+ * @details Given methods are set as list of methods for the given configuration.
+ * Caller is response for allocating and deallocating memory for both configuration and method list.
+ * 'methods' memory area must be kept allocated as long as the configuration is used.
+ *
+ * @param[inout] config Configuration which is supplied with default values.
+ * @param[in] methods_count Number of methods to be stored in 'methods'.
+ * @param[in] methods List of methods. This will be initialized to zero.
+ */
+void loc_config_defaults_set(struct loc_config *config, uint8_t methods_count,
+			     struct loc_method_config *methods);
+
+/**
+ * @brief Sets default values for given method configuration based on given type.
+ *
+ * @details Intended use is that this is part of loc_config that has been initialized earlier.
```

Comment:
If it is part of loc_config initialized earlier, does that mean this is for re-configuring an already configured method?

Reply:
Not really. This is for doing initial configuration for the specific location method. Of course, you can run this again to reconfigure the method and then use the parent location config another time but I'm not expecting this to be a use case for applications.
You can see how this is used in modem shell at: https://github.com/tokangas/sdk-nrf/blob/location_api/samples/nrf9160/modem_shell/src/location/location_shell.c#L356

Reply:
Maybe word "earlier" is not good in this description as it refers to few lines above rather than seconds, minutes or hours earlier.

Reply:
Pushed a fix to this